picard_collectmultiplemetric_2.8.1/picard_collectmultiplemetrics_2-8-1.cwl

```diff
@@ -0,0 +1,215 @@
+  - id: metric_accumulation_level
+    type: string?
+    inputBinding:
+      position: 0
+      prefix: METRIC_ACCUMULATION_LEVEL=
+      separate: false
+    doc: >-
+      The level(s) at which to accumulate metrics. Default value: [ALL_READS].
+      This option can be set to 'null' to clear the default value. Possible
+      values: {ALL_READS, SAMPLE, LIBRARY, READ_GROUP} This option may be
+      specified 0 or more times. This option can be set to 'null' to clear the
+      default list.
+  - id: file_extension
+    type: string?
+    inputBinding:
+      position: 0
+      prefix: FILE_EXTENSION=
+      separate: false
+    doc: >-
+      Append the given file extension to all metric file names (ex.
+      OUTPUT.insert_size_metrics.EXT). None if null Default value: null.
+  - id: program_list
+    type: 'string[]?'
+    inputBinding:
+      position: 0
+      prefix: PROGRAM=
+      separate: false
+    doc: >-
+      Set of metrics programs to apply during the pass through the SAM file.
+      Default value: [CollectAlignmentSummaryMetrics,
+      CollectBaseDistributionByCycle, CollectInsertSizeMetrics,
+      MeanQualityByCycle, QualityScoreDistribution]. This option can be set to
+      'null' to clear the default value. Possible values:
+      {CollectAlignmentSummaryMetrics, CollectInsertSizeMetrics,
+      QualityScoreDistribution, MeanQualityByCycle,
+      CollectBaseDistributionByCycle, CollectGcBiasMetrics, RnaSeqMetrics,
+      CollectSequencingArtifactMetrics, CollectQualityYieldMetrics} This option
+      may be specified 0 or more times. This option can be set to 'null' to
+      clear the default list.
+  - id: intervals_file
+    type: File?
+    inputBinding:
+      position: 0
+      prefix: INTERVALS=
+      separate: false
+    doc: >-
+      An optional list of intervals to restrict analysis to. Only pertains to
+      some of the PROGRAMs. Programs whose stand-alone CLP does not have an
+      INTERVALS argument will silently ignore this argument. Default value:
+      null.
+  - id: dbsnp_file
+    type: File?
+    inputBinding:
+      position: 0
+      prefix: DB_SNP=
+      separate: false
+    doc: >-
+      VCF format dbSNP file, used to exclude regions around known polymorphisms
+      from analysis by some PROGRAMs; PROGRAMs whose CLP doesn't allow for this
+      argument will quietly ignore it. Default value: null.
+  - id: include_unpaired
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: INCLUDE_UNPAIRED=true
+      separate: false
+    doc: >-
+      Include unpaired reads in CollectSequencingArtifactMetrics. If set to true
+      then all paired reads will be included as well - MINIMUM_INSERT_SIZE and
+      MAXIMUM_INSERT_SIZE will be ignored in CollectSequencingArtifactMetrics.
+      Default value: false. This option can be set to 'null' to clear the
+      default value. Possible values: {true, false}
+outputs:
+  - id: all_metrics
+    type: Directory
```

Comment:
Why is this a directory, shouldn't we get all the files related to a basename?

Reply:
@ionox0 and I discussed this and decided to change it from basename to Directory. @ionox0 Do you want to comment on this?

Reply:
https://gitlab.ddbj.nig.ac.jp/genomics-workflow-standards/human-reseq/blob/734dc99af9d2ea36d1891753847408576449a74d/Tools/picard-CollectMultipleMetrics.cwl